unlock-app/src/__tests__/utils/metadataMunging.test.ts

```diff
@@ -0,0 +1,149 @@
+}
+
+describe('metadata munging functions', () => {
+  describe('generateColumns', () => {
+    it('returns the starting columns when given empty input', () => {
+      expect.assertions(2)
+
+      let columns = generateColumns([])
+      expect(columns).toEqual(['lockName', 'keyholderAddress', 'expiration'])
+
+      columns = generateColumns([], ['favoriteColor'])
+      expect(columns).toEqual(['favoriteColor'])
+    })
+
+    it('places the starting columns first and then alphabetizes additional columns', () => {
+      expect.assertions(1)
+      const keyMetadata = {
+        ...baseKeyMetadata,
+        zebras: 'true',
+        platypus: 'weird',
+        apples: 'oranges',
+        beagles: 'doggo',
+      }
+
+      const columns = generateColumns([keyMetadata])
+      expect(columns).toEqual([
+        'lockName',
+        'keyholderAddress',
+        'expiration',
+        // Alphabetizing starts here, after the starting columns
+        'apples',
+        'beagles',
+        'platypus',
+        'zebras',
+      ])
+    })
+
+    it('pulls columns from multiple pieces of data', () => {
+      expect.assertions(1)
+      const firstKey = {
+        ...baseKeyMetadata,
+        email: 'in@ter.net',
+      }
+      const secondKey = {
+        ...baseKeyMetadata,
+        xylophone: 'a value',
+      }
+
+      const columns = generateColumns([firstKey, secondKey])
+      expect(columns).toEqual([
+        'lockName',
+        'keyholderAddress',
+        'expiration',
+        'email',
+        'xylophone',
+      ])
+    })
+  })
+
+  describe('mergeSingleDatum', () => {
+    it('merges a base metadatum with additional fields from stored metadata', () => {
+      expect.assertions(1)
+      const lock = {
+        address: '0x123',
+        name: 'Lock',
+      }
+      const key = {
+        keyId: '1',
+        expiration: '123456',
+        owner: {
+          address: '0x123',
+        },
+      }
+      const storedMetadata = {
+        [lock.address]: {
+          [key.keyId]: {
+            protected: {
+              zebra: 'true',
+            },
+            public: {
+              elephant: 'trunk',
+            },
+          },
+        },
+      }
+      const metadatum = mergeSingleDatum(lock, key, storedMetadata)
+      expect(metadatum).toEqual({
+        lockName: 'Lock',
+        expiration: 'Expired',
+        keyholderAddress: '0x123',
+        elephant: 'trunk',
+        zebra: 'true',
+      })
+    })
+  })
+
+  describe('mergeKeyholderMetadata', () => {
+    it('returns empty array when there are no locks', () => {
+      expect.assertions(1)
+      expect(
+        mergeKeyholderMetadata(
+          {
+            locks: [],
+          },
+          {}
+        )
+      ).toEqual([])
+    })
+
+    it('adds metadata for each key in a lock', () => {
+      expect.assertions(1)
+      const data = {
+        locks: [
+          {
+            address: '0x123',
+            name: 'Lock',
+            keys: [
```

Comment:
It would be a good idea to update this test with an additional key to ensure that we handle the multiple key case